onboarding/main.go

```diff
@@ -57,90 +56,87 @@ func main() {
+		},
+	}
+	err = cl.Delete(ctx, publicSecret)
 	if err != nil && !kerrors.IsNotFound(err) {
 		klog.Errorf("failed to delete public secret: %v", err)
 		os.Exit(1)
 	}
 
-	storageClusterMetadata, err := getStorageClusterMetadata(ctx, operatorNamespace, clientset)
+	storageClusterMetadata, err := getStorageClusterMetadata(ctx, operatorNamespace, cl)
 	if err != nil {
 		klog.Errorf("failed to get storage cluster metadata: %v", err)
 		os.Exit(1)
 	}
 
-	privateSecret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      onboardingValidationPrivateKeySecretName,
-			Namespace: operatorNamespace,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					UID:        storageClusterMetadata.UID,
-					APIVersion: storageClusterMetadata.APIVersion,
-					Kind:       storageClusterMetadata.Kind,
-					Name:       storageClusterMetadata.Name,
-				},
-			}},
-		StringData: map[string]string{
-			"key": privatePem,
+	privateSecret.ObjectMeta.OwnerReferences = []metav1.OwnerReference{
+		{
+			UID:        storageClusterMetadata.UID,
+			APIVersion: storageClusterMetadata.APIVersion,
+			Kind:       storageClusterMetadata.Kind,
+			Name:       storageClusterMetadata.Name,
 		},
 	}
+	privateSecret.StringData = map[string]string{
+		"key": privatePem,
+	}
 
-	_, err = clientset.CoreV1().
-		Secrets(operatorNamespace).
-		Create(ctx, privateSecret, metav1.CreateOptions{})
-
+	err = cl.Create(ctx, privateSecret, &client.CreateOptions{})
 	if err != nil {
 		klog.Errorf("failed to create private secret: %v", err)
 		os.Exit(1)
 	}
 
-	publicSecret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      onboardingValidationPublicKeySecretName,
-			Namespace: operatorNamespace,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					UID:        storageClusterMetadata.UID,
-					APIVersion: storageClusterMetadata.APIVersion,
-					Kind:       storageClusterMetadata.Kind,
-					Name:       storageClusterMetadata.Name,
-				},
-			}},
-		StringData: map[string]string{
-			"key": publicPem,
+	publicSecret.ObjectMeta.OwnerReferences = []metav1.OwnerReference{
+		{
+			UID:        storageClusterMetadata.UID,
+			APIVersion: storageClusterMetadata.APIVersion,
+			Kind:       storageClusterMetadata.Kind,
+			Name:       storageClusterMetadata.Name,
 		},
 	}
+	publicSecret.StringData = map[string]string{
+		"key": publicPem,
+	}
 
-	_, err = clientset.CoreV1().
-		Secrets(operatorNamespace).
-		Create(ctx, publicSecret, metav1.CreateOptions{})
+	err = cl.Create(ctx, publicSecret, &client.CreateOptions{})
 	if err != nil {
 		klog.Errorf("failed to create public secret: %v", err)
 		os.Exit(1)
 	}
 
 }
 
-func newClient() (*kubernetes.Clientset, error) {
-	config := runtime.GetConfigOrDie()
-	clientset, err := kubernetes.NewForConfig(config)
+func newClient() (client.Client, error) {
+	klog.Info("Setting up k8s client")
+	config, err := config.GetConfig()
 	if err != nil {
-		return nil, err
+		klog.Exitf("Failed to get config: %v", err)
```

Comment:
Don't exist from within a util function. Exit should only be used at the top-level function (e.g. main). Return an error instead